internal/service/customdbrole/resource_custom_db_role.go

```diff
@@ -173,55 +165,42 @@ func resourceMongoDBAtlasCustomDBRoleRead(ctx context.Context, d *schema.Resourc
 		return diag.FromErr(fmt.Errorf("error getting custom db role information: %s", err))
 	}
 
-	if err := d.Set("role_name", customDBRole.RoleName); err != nil {
+	if err := d.Set("role_name", customDBRole.GetRoleName()); err != nil {
 		return diag.FromErr(fmt.Errorf("error setting `role_name` for custom db role (%s): %s", d.Id(), err))
 	}
 
-	if err := d.Set("actions", flattenActions(customDBRole.Actions)); err != nil {
+	if err := d.Set("actions", flattenActions(customDBRole.GetActions())); err != nil {
 		return diag.FromErr(fmt.Errorf("error setting `actions` for custom db role (%s): %s", d.Id(), err))
 	}
 
-	if err := d.Set("inherited_roles", flattenInheritedRoles(customDBRole.InheritedRoles)); err != nil {
+	if err := d.Set("inherited_roles", flattenInheritedRoles(customDBRole.GetInheritedRoles())); err != nil {
 		return diag.FromErr(fmt.Errorf("error setting `inherited_roles` for custom db role (%s): %s", d.Id(), err))
 	}
 
 	return nil
 }
 
-func resourceMongoDBAtlasCustomDBRoleUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	customRoleLock.Lock()
-	defer customRoleLock.Unlock()
```

Comment:
Nice cleanup, understand this has no value as is it global state outside of CRUD operations, and also not clear why a lock is needed.

Reply:
correct, i think this is maybe a workaround done some  time ago because some transient problem but it's not needed.  i haven't found this hack in other places